docs/highlighters/regexp.md

```diff
@@ -28,3 +28,19 @@ in [the `zshmisc(1)` manual page][zshmisc-Conditional-Expressions]
 [zshzle-Character-Highlighting]: http://zsh.sourceforge.net/Doc/Release/Zsh-Line-Editor.html#Character-Highlighting
 [perlretut]: http://perldoc.perl.org/perlretut.html
 [zshmisc-Conditional-Expressions]: http://zsh.sourceforge.net/Doc/Release/Conditional-Expressions.html#Conditional-Expressions
+
+### Cautions to BSD-based platform users
```

Comment:
As written, this could make "BSD-based platform" users feel like second-class citizens, because the example in the manual doesn't work for them.  Adding a section that documents how to adapt the example in the actual documentation to one's platform is less than ideal: see https://m.xkcd.com/1343/.

I would prefer to give a single example that works everywhere, or failing that, to give several examples next to each other, putting all supported platforms on equal footing.  Makes sense?